llvm/include/llvm/MC/MCInstrAnalysis.h

```diff
@@ -52,6 +53,10 @@ class MCInstrAnalysis {
   /// new function), clients should call resetState to clear it.
   virtual void updateState(const MCInst &Inst, uint64_t Addr) {}
 
+  /// Let the analysis update disassembled instruction from a recorded internal
+  /// state.
```

Comment:
Analyses generally shouldn't mutate what they are analyzing?

Reply:
Agreed. In this case I think the justification is that the mutation just records some helpful extra info in the MCInst that would not affect the encoding of the instruction (and I guess we could verify that).

We did also discuss offline a way to avoid this altogether, by getting the target InstPrinter::printOperand to query the MCInstrAnalysis directly.

Reply:
Right, the modification merely will add extra info for comment printing. That way the MCInst will be produced consistently with the way codegen does it, and will be handled the same way in the InstPrinter, so no special casing is needed.

I can write an extra code in the asm printer for the case when code comes from disasm and not from codegen, it is just less clean to me. It also would require to make AMDGPUMCInstrAnalysis a class with public interface and move it out of the anonymous namespace. These are the reasons I'd prefer to add that extra info to the MCInst itself.